app/components/sidebars/main/channels_list/channel_item/index.js

```diff
@@ -27,15 +26,13 @@ function makeMapStateToProps() {
 
         let isMyUser = false;
         let teammateDeletedAt = 0;
-        let displayName = channel.display_name;
-        if (channel.type === General.DM_CHANNEL && channel.teammate_id) {
-            isMyUser = channel.teammate_id === currentUserId;
-            const teammate = getUser(state, channel.teammate_id);
-            if (teammate && teammate.delete_at) {
-                teammateDeletedAt = teammate.delete_at;
-            }
-            const teammateNameDisplay = getTeammateNameDisplaySetting(state);
-            displayName = displayUsername(teammate, teammateNameDisplay, false);
+        let isArchived = false;
+        const displayName = channel.display_name;
+        if (channel.type === General.DM_CHANNEL) {
+            isMyUser = channel.id === currentUserId;
+            teammateDeletedAt = channel.delete_at;
```

Comment:
I don't think `delete_at` of teammate (user) can be safely inferred as from their DM channel.  I'm confused, maybe please clarify.

Reply:
These values are coming from https://github.com/mattermost/mattermost-mobile/pull/2080/files#diff-23e5d2f83331fd7fccf6be48c0aa6052R260 and https://github.com/mattermost/mattermost-mobile/pull/2080/files#diff-23e5d2f83331fd7fccf6be48c0aa6052R218

We are making these objects in parent component with few extra keys like `diplay_name` and we do have the data for delete_at as well so adding the key there

Reply:
That FilteredList component needs to be re-written anyways, it is actually very poor performant, I think some work is being done with the sidebar re-org by @jarredwitt maybe he can confirm.